src/screens/DiscoveryScreenV2/fragment/UsersFragment.js

```diff
@@ -92,72 +90,74 @@ const UsersFragment = ({
-        setTimeout(() => {
-          exhangeFollower(newFollowedUsers, !willFollow, index);
-          DiscoveryAction.setDiscoveryInitialUsers(newFollowedUsers, discoveryDispatch);
-        }, 2000);
+      DiscoveryAction.setDiscoveryInitialUsers(
+        discovery.initialUsers.map((user) => {
+          if (user.user) {
+            if (user.user.user_id === newUser.user.user_id) return newUser;
+          } else if (user.user_id === newUser.user_id) return newUser;
+          return user;
+        }),
+        discoveryDispatch
+      );
     }
 
     if (from === FROM_FOLLOWED_USERS) {
       const newFollowedUsers = [...followedUsers];
       exhangeFollower(newFollowedUsers, willFollow, item.user ? item.user.user_id : item.user_id);
 
       setFollowedUsers(newFollowedUsers);
-      if (!netInfo.isConnected)
-        setTimeout(() => {
-          exhangeFollower(newFollowedUsers, !willFollow, index);
-          setFollowedUsers(newFollowedUsers);
-        }, 2000);
     }
 
     if (from === FROM_UNFOLLOWED_USERS) {
       const newUnfollowedUsers = [...unfollowedUsers];
       exhangeFollower(newUnfollowedUsers, willFollow, item.user ? item.user.user_id : item.user_id);
       setUnfollowedUsers(newUnfollowedUsers);
-      if (!netInfo.isConnected)
-        setTimeout(() => {
-          exhangeFollower(newUnfollowedUsers, !willFollow, index);
-          setUnfollowedUsers(newUnfollowedUsers);
-        }, 2000);
     }
 
     if (from === FROM_USERS_INITIAL) {
       const newFollowedUsers = [...initialUsers];
       exhangeFollower(newFollowedUsers, willFollow, item.user ? item.user.user_id : item.user_id);
 
       setInitialUsers(newFollowedUsers);
-      if (!netInfo.isConnected)
-        setTimeout(() => {
-          exhangeFollower(newFollowedUsers, !willFollow, index);
-          setInitialUsers(newFollowedUsers);
-        }, 2000);
     }
+  };
 
-    if (netInfo.isConnected) {
-      const data = {
-        user_id_follower: myId,
-        user_id_followed: item.user_id,
-        username_follower: profile.myProfile.username,
-        username_followed: item.username,
-        follow_source: 'discoveryScreen'
-      };
+  const handleFollow = async (from, willFollow, item) => {
+    handleUser(from, willFollow, item);
+    const data = {
+      user_id_follower: myId,
+      user_id_followed: item.user ? item.user.user_id : item.user_id,
+      username_follower: profile.myProfile.username,
+      username_followed: item.username,
+      follow_source: 'discoveryScreen'
+    };
 
-      if (willFollow) {
+    if (willFollow) {
+      try {
         await setFollow(data, client);
-      } else {
+      } catch (error) {
+        handleUser(from, !willFollow, item);
+      }
+    } else {
+      try {
         await setUnFollow(data, client);
+      } catch (error) {
+        handleUser(from, !willFollow, item);
       }
-      if (searchText.length > 0) fetchData();
     }
+    if (searchText.length > 0) fetchData();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new `handleUser` function seems to be a refactoring of the previous logic in `handleFollow`. However, it appears that the handling for when the device is not connected to the internet (previously managed with `useNetInfo`) has been removed. This could lead to unexpected behavior when the device is offline. 

Also, the `handleFollow` function now calls `handleUser` without awaiting its completion. Since `handleUser` is an asynchronous function, this could potentially lead to race conditions if `handleUser` hasn't finished executing before the next lines of code are run.

Consider adding back the network connectivity check and await the `handleUser` function call in `handleFollow`.
```diff
-    handleUser(from, willFollow, item);
+    await handleUser(from, willFollow, item);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->